src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -477,20 +477,17 @@ private static unsafe void Avx2Decode(ref byte* srcBytes, ref byte* destBytes, b
             destBytes = dest;
         }
 
-        // This can be replaced once https://github.com/dotnet/runtime/issues/63331 is implemented.
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static Vector128<byte> SimdShuffle(Vector128<byte> left, Vector128<byte> right, Vector128<byte> mask8F)
         {
             Debug.Assert((Ssse3.IsSupported || AdvSimd.Arm64.IsSupported) && BitConverter.IsLittleEndian);
 
-            if (Ssse3.IsSupported)
+            if (AdvSimd.Arm64.IsSupported)
             {
-                return Ssse3.Shuffle(left, right);
-            }
-            else
-            {
-                return AdvSimd.Arm64.VectorTableLookup(left, Vector128.BitwiseAnd(right, mask8F));
+                right &= mask8F;
```

Comment:
honestly speaking I'm not sure this operation is even needed here for base64 but it's irrelevant to this PR

Reply:
cc: @gfoidl would you mind checking if that's the case? Looks like the indices are already being masked by the callers in a few ways, but I am not familiar enough with the base64 logic here.
If it is redundant, we can remove the `SimdShuffle` helper entirely.

Reply:
I bet it was added during porting that algorithm to ARM64 and to adjust `VectorTableLookup` to behave like SSSE3 Shuffle in a general case

Reply:
If ARM for the `VectorTableLookup` only takes the low nibble into account (I don't know) it's not needed, and after https://github.com/dotnet/runtime/pull/76795 there should be enough tests to cover non-ASCII input cases well.
In the code the masking of the input-nibbles happens with `0x2F` https://github.com/dotnet/runtime/blob/c2b12d7fce16b9e4ec3a44bf747118732e85782c/src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs#L591-L592 instead of `0xF`, as that constant already is around thus saving the use of another register. Then after the check for invalid input, only valid values are around, so further down the code no bits in the high-nibble exist.

If `VectorTableLookup` takes the higher nibble into account, then masking with `0xF` instead of `0x8F` is fine.

> behave like SSSE3 Shuffle in a general case

This seems very plausible, as `0x8F` has the highest bit set with has special meaning in SSSE3 shuffle.